pkg/receive/hashring.go

```diff
@@ -6,14 +6,31 @@ package receive
 import (
 	"context"
 	"fmt"
+	"sort"
+	"strconv"
 	"sync"
 
+	"github.com/cespare/xxhash"
+
 	"github.com/pkg/errors"
 	"github.com/thanos-io/thanos/pkg/store/labelpb"
 
 	"github.com/thanos-io/thanos/pkg/store/storepb/prompb"
 )
 
+// HashringAlgorithm is the algorithm used to distribute series in the ring.
+type HashringAlgorithm string
+
+const (
+	AlgorithmHashmod    HashringAlgorithm = "hashmod"
+	AlgorithmConsistent HashringAlgorithm = "consistent"
```

Comment:
I need to agree with @douglascamata I think it would quite good to specify exactly what implementation we use. If it's based on libketama from https://dgryski.medium.com/consistent-hashing-algorithmic-tradeoffs-ef6b8e2fcae8 - let's mentioned that in comment, name of this `ketama`, and rename structs.

Reply:
👍